### PR TITLE
Add type checking in Where Do I Belong

### LINKED ISSUE #13493: [Where do I belong] Right results but not pass the test
Challenge [Where do I belong](https://www.freecodecamp.com/challenges/where-do-i-belong#?solution=%0Afunction%20getIndexToIns(arr%2C%20num)%20%7B%0A%20%20arr%20%3D%20arr%3B%0A%20%20for%20(var%20i%20in%20arr.sort(function(a%2Cb)%7Breturn%20a-b%3B%7D))%7B%0A%20%20%20%20if%20(arr%5Bi%5D%20%3E%3D%20num)%20%7Breturn%20i%3B%7D%0A%20%20%7D%0A%20%20return%20arr.length%3B%0A%7D%0A%0AgetIndexToIns(%5B3%2C%2010%2C%205%5D%2C%203)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

function getIndexToIns(arr, num) {
  arr = arr;
  for (var i in arr.sort(function(a,b){return a-b;})){
    if (arr[i] >= num) {return i;}
  }
  return arr.length;
}

getIndexToIns([3, 10, 5], 3);

```

I got in the return all the right value from the test, it just don't get it right.

### PR DESCRIPTION
**Pull request by Manish-Giri**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #13493  

#### Description

Added tests to verify that a `number` was returned. After this change - 

OP's code does not pass - 
```
function getIndexToIns(arr, num) {
  arr = arr;
  for (var i in arr.sort(function(a,b){return a-b;})){
    if (arr[i] >= num) {return i;}
  }
  return arr.length;
}

getIndexToIns([3, 10, 5], 3);
```
screenshot 2017-02-25 19 59 03

While this code does - 
```
function getIndexToIns(arr, num) {
  // Find my place in this sorted array.
  arr.push(num);
  arr.sort(function(a,b){return a-b;});
  return arr.indexOf(num);
}
```
screenshot 2017-02-25 19 59 43

Note - I did think about merging the two tests for each case into one, but that is known to [not work correctly before](https://github.com/freeCodeCamp/freeCodeCamp/pull/11912#issuecomment-266926761). 

Closes #13493

### CODE REVIEW

**Comment by erictleung** [comment]
@Manish-Giri hey there! Sorry for the late QA. For some reason, I'm unable to properly test this. Does this still work for you? There were some recently changes with nuking the `bower.json` file which is preventing me from working on this. It might be more of my issue, but just wondering if you ran into it. Thanks again for all the PRs!

**Comment by Manish-Giri** [comment]
@erictleung I just created a new branch against the latest staging environment, and it still works. Should I create a new PR with the latest branch, just in case?

**Comment by erictleung** [comment]
@Manish-Giri mmm sounds like more of my problem that I can't QA this. Let me try again this weekend. If others are able to QA this before I am, then feel free.

**Comment by raisedadead** [comment]
@erictleung You may have to nuke `public/bower_components`, to resolve any conflicts.

**Review by erictleung — APPROVED** [review]
Ah got my development environment working to QA this finally. LGTM! 👍  Awesome work! Also, slick solution you have to pass the challenge. Very clever 😄

**Comment by Manish-Giri** [comment]
@erictleung Thanks for QAing this, and thanks for this too :smile: 
> Also, slick solution you have to pass the challenge. Very clever
